internal/vm/ioproxy.go

```diff
@@ -152,8 +152,8 @@ func (connectorPair *IOConnectorPair) proxy(
 			} else {
 				logger.WithError(err).Error("error copying io")
 			}
-			copyDone <- err
 		}
+		copyDone <- err
```

Comment:
qq: we close the `copyDone` channel regardless of whether the copy succeeds or fails. Shouldn't that indicate the runtime to clean up the ioproxy?

Reply:
Ah, yes. I missed that. Using the debug logs from https://github.com/firecracker-microvm/firecracker-containerd/pull/766 we actually don't see the "copied %d" message from https://github.com/firecracker-microvm/firecracker-containerd/blob/fc5e81ef584b54e91f7d7f8f6a1432dead7fa060/internal/vm/ioproxy.go#L147 which seems like we're actually still stuck in the `io.CopyBuffer`?

Reply:
Do we see the `copied` message with your change to explicitly send an error?

[this](https://github.com/firecracker-microvm/firecracker-containerd/blob/main/internal/vm/ioproxy.go#L209-L211) is where we wait for an error in the copy channel. I assumed this should return nil once we close the channel.